android/source/VideoLocker/src/org/edx/mobile/view/registration/RegistrationEditTextView.java

```diff
@@ -0,0 +1,69 @@
+package org.edx.mobile.view.registration;
+
+import android.text.InputFilter;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+
+import org.edx.mobile.R;
+import org.edx.mobile.model.registration.RegistrationFormField;
+
+public class RegistrationEditTextView implements IRegistrationFieldView {
+    private RegistrationFormField mField;
+    private View mView;
+    private EditText mInputView;
+    private TextView mErrorView, mInstructionView;
+
+    public RegistrationEditTextView(RegistrationFormField field, View view) {
+        // create and configure view and save it to an instance variable
+        this.mField = field;
+        this.mView = view;
+
+        this.mInputView = (EditText) view.findViewById(R.id.txt_input);
+        this.mErrorView = (TextView) view.findViewById(R.id.txt_input_error);
+        this.mInstructionView = (TextView) view.findViewById(R.id.txt_input_instruction);
+
+        // apply max length
+        InputFilter[] FilterArray = new InputFilter[1];
+        FilterArray[0] = new InputFilter.LengthFilter(mField.getRestriction().getMaxLength());
+        mInputView.setFilters(FilterArray);
```

Comment:
@aleffert @nasthagiri Here we don't allow typing more characters than <code>max_length</code>. Rather, should we allow typing more chars and later during validation, show the error message?